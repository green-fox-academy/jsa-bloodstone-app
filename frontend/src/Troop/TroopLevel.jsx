import React from 'react';
import PropTypes from 'prop-types';
import {
  View, Image, Text,
  StyleSheet, TouchableHighlight,
} from 'react-native';
import commonStyles from '../common/styles';
import troopRound from '../../assets/troop/troop.png';

const styles = StyleSheet.create({
  roundAvatar: {
    width: 60,
    height: 60,
    marginRight: 12,
  },
});

function TroopLevel({ troops, level }) {
  function handlePress() {
    // Todo
  }

  return (
    <TouchableHighlight underlayColor="transparent" onPress={handlePress}>
      <View style={{ flexDirection: 'row', justifyContent: 'center' }}>
        <View style={commonStyles.cardStyle}>
          <Image source={troopRound} style={styles.roundAvatar} />
          <Text style={commonStyles.cardText}>
            {troops}
            Troop level
            {level}
          </Text>
        </View>
      </View>
    </TouchableHighlight>
  );
}

TroopLevel.propTypes = {
  level: PropTypes.number.isRequired,
  troops: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default TroopLevel;
