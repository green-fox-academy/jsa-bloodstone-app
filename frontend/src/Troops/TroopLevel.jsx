import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  Animated,
  View, Image, Text,
  StyleSheet, TouchableHighlight,
} from 'react-native';
import Colors from '../common/colors';
import troopRound from '../../assets/troop/troop.png';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginBottom: 8,
    alignItems: 'center',
  },
  roundAvatar: {
    width: 40,
    height: 40,
    marginRight: 12,
  },
  cardStyle: {
    flex: 1,
    padding: 10,
    borderRadius: 10,
    flexDirection: 'row',
    backgroundColor: Colors.tealColor,
  },
  cardText: {
    color: '#fff',
    fontSize: 16,
  },
  upgradeStyle: {
    width: 60,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
    borderTopEndRadius: 5,
    borderBottomEndRadius: 5,
    backgroundColor: Colors.redColor,
  },
  upgradeText: {
    fontSize: 8,
    color: '#fff',
  },
});

function TroopLevel({ count, level }) {
  const [showUpgradeButton, setShowUpgradeButton] = useState(false);
  const [widthOfUpgrade] = useState(new Animated.Value(0));

  function handlePress() {
    setShowUpgradeButton(!showUpgradeButton);
    Animated.timing(
      widthOfUpgrade,
      {
        toValue: showUpgradeButton ? 0 : 50,
        duration: 200,
      },
    ).start();
  }

  return (
    <View style={styles.container}>
      <TouchableHighlight
        style={styles.cardStyle}
        onPress={handlePress}
        underlayColor={Colors.tealColor}
      >
        <View style={{ flexDirection: 'row', alignItems: 'center' }}>
          <Image source={troopRound} style={styles.roundAvatar} />
          {}
          <Text style={styles.cardText}>
            {`Level ${level}: ${count} Troop${count > 1 ? 's' : ''}`}
          </Text>
        </View>
      </TouchableHighlight>
      <Animated.View style={{ ...styles.upgradeStyle, width: widthOfUpgrade }}>
        <TouchableHighlight underlayColor="transparent" onPress={() => null}>
          <Text numberOfLines={1} style={styles.upgradeText}>UPGRADE</Text>
        </TouchableHighlight>
      </Animated.View>
    </View>
  );
}

TroopLevel.propTypes = {
  level: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
};

export default TroopLevel;
