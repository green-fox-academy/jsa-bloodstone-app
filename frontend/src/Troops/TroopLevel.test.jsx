import React from 'react';
import renderer from 'react-test-renderer';

import TroopLevel from './TroopLevel';

jest.mock('react-redux');
jest.mock('./actionCreator');

describe('<TroopLevel />', () => {
  it('Should render troop level and match the snapshot', () => {
    const component = renderer.create(<TroopLevel />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
