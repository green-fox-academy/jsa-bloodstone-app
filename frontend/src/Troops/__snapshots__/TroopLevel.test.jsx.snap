// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TroopLevel /> Should render troop level and match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 8,
      "marginHorizontal": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#10978b",
        "borderRadius": 10,
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 40,
            "marginRight": 12,
            "width": 40,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
          }
        }
      >
        Level undefined: undefined Troops
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f44336",
        "borderBottomEndRadius": 5,
        "borderTopEndRadius": 5,
        "height": 40,
        "justifyContent": "center",
        "textAlign": "center",
        "width": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={null}
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#fff",
            "fontSize": 8,
          }
        }
      >
        UPGRADE
      </Text>
    </View>
  </View>
</View>
`;
